src/components/organisms/NewProfileModal/components/ProfileModalFetchUser/ProfileModalFetchUser.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { FalseyValue } from "styled-components";
+
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+
+import { Loading } from "components/molecules/Loading";
+
+import { useCurrentModalUser } from "./useCurrentModalUser";
+
+import "./ProfileModalFetchUser.scss";
+
+export interface ProfileModalFetchUserProps {
+  userId?: string;
+  children: (user: WithId<User>) => React.ReactElement | FalseyValue | "";
+}
+
+export const ProfileModalFetchUser: React.FC<ProfileModalFetchUserProps> = ({
+  userId,
+  children,
+}: ProfileModalFetchUserProps) => {
+  const [user, isLoaded] = useCurrentModalUser(userId);
+
+  if (isLoaded && user) return <>{children(user)}</>;
+
+  if (!isLoaded)
+    return (
+      <div className="ProfileModalFetchUser">
+        <Loading />
+      </div>
+    );
+
+  if (!user)
+    return (
+      <div className="ProfileModalFetchUser">
+        Oops, an error occurred while trying to load user data.{"\n"}
+        Please contact our support team.
+      </div>
+    );
+
+  return <></>;
+};
```

Comment:
This component looks a bit redundant/new for me. I do understand that you wanted to encapsulate loading/error state for user, but probably it's better to do via hook, not the component. @sunny-viktoryia @denisdimitrov wdyt of this approach?

Reply:
in fact, it was me who suggested refactoring the component this way - the previous version of it seemed overcomplicated.

I'd just move `<>{children(user)}</>` at the end (instead of empty `<></>`)  - as if we reach it, it means that `isLoaded && user` is truth.

But I don't mind having the logic in a hook as well.

Reply:
removed in https://github.com/sparkletown/sparkle/pull/2291/commits/2fe56537774a736c8a86028c565bb49da1b08801